### PR TITLE
Add mod downloader + improve codebase abstraction + add concurrency to operations

### PR DESCRIPTION
**Pull request by SwadicalRag**

### CODE REVIEW

File: update-mods-md.ts [hunk 1]
```diff
@@ -229,11 +83,28 @@ async function main(argc: number, argv: string[])
...
-		const newTable = await processMods(table);
-		newLines.push(formatModTable(newTable, columnWidths));
+		updateModIDs(concurrency,table);
+		newLines.push([nextI, table, columnWidths]);
 		i = nextI;
 	}
 
+	await concurrency.defer();
+
+	for(let i=0; i < newLines.length;i++)
```
Comment (notcake) [inline update-mods-md.ts]: The whitespace makes me sad >:

**Review by notcake — CHANGES_REQUESTED** [review]

File: update-mods-md.ts [hunk 2]
```diff
@@ -205,14 +56,17 @@ async function main(argc: number, argv: string[])
 
 	const data = fs.readFileSync(process.argv[2], "utf-8");
 	const lines = data.split("\n");
-	const newLines: string[] = [];
+	const newLines: (string | [number,string[][],number[]])[] = [];
```
Comment (notcake) [inline update-mods-md.ts:59]: whitespace

File: update-mods-md.ts [hunk 3]
```diff
@@ -229,11 +83,28 @@ async function main(argc: number, argv: string[])
 			continue;
 		}
 
-		const newTable = await processMods(table);
-		newLines.push(formatModTable(newTable, columnWidths));
+		updateModIDs(concurrency,table);
```
Comment (notcake) [inline update-mods-md.ts:86]: whitespace

**Review by notcake — APPROVED** [review]

**Comment by SwadicalRag** [comment]
im so sorry q_q

**Pull request merged**